### ISSUE TITLE
[Bug]: No information about currently selected date in Calendar for Screen reader user

### ISSUE THREAD

**Issue by satyaminnekanti** [carbon-design-system/carbon#16247]
### Package

carbon-components

### Browser

Chrome

### Package version

11

### Description

JAWS reads the date and also informs that its selected .

### Reproduction/example

https://carbondesignsystem.com/components/date-picker/usage/#live-demo

### Steps to reproduce

- Go to [Date Picker live demo](https://carbondesignsystem.com/components/date-picker/usage/#live-demo)
- select a date
- with screen reader ON, Open the date picker and navigate to the selected date

**Expected Result**

JAWS reads the date and also informs that its selected 

**Actual Result**

JAWS just reads the date **BUT no information about the selected state. This is a failure of WCAG [SC 4.1.2:Name, Role, Value](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value.html)**

Please refer [ARIA Authoring Practices date picker example](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-datepicker/)

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by kennylam** [comment]
@maradwan26 to follow-up, suspect fix has already been done.

**Comment by maradwan26** [comment]
Can confirm that this issue still exists. Need more accessibility info on what the remediation should be:
- `aria-selected` cannot be used here since the elements have role="button"
- `aria-pressed` will result in the dates being announced as toggle buttons which isn't correct
- Switching to `grid` and `grid-cell` roles and using `aria-selected` will announce but alongside row/column info which can be verbose
    - Implementing may also have some resistance from FlatPickr?
- ...

**Comment by chrisgrimley** [comment]
Hi @satyaminnekanti and @maradwan26. Similar to the open [issue with the tab index on the DatePicker](https://github.com/carbon-design-system/carbon/issues/22579) component, I believe this is another accessibility limitation of Flatpickr. It appears to have many incomplete ARIA attributes that would address this (e.g. role on the day, aria-selected on the chosen date). Rather than patch this, I'd suggest ensuring this works in the [V12 DatPicker](https://labs.carbondesignsystem.com/v12/?path=/docs/web-components_v12-date-picker--docs#simple-date-picker) and migrating when it's Production-ready

**Comment by maradwan26** [comment]
@chrisgrimley 
Thanks for your input! I'll move this issue under the preview_DatePicker migration/stabilization Epic. Just a question, though, is switching to `grid` and `grid-cell` roles with `aria-selected` the recommended approach here? Or is there a better way?

**Comment by chrisgrimley** [comment]
Hi @maradwan26, apologies for only seeing your question above. Yes, the grid and gridcell roles seem to be the suggested approach in the W3C APG:

```
<td role="gridcell"
      aria-selected="true"
      data-date="2020-02-14">
             14
</td> 
```
For comparison, I also looked up to see what Material Design do and they appear to also use this approach, but interestingly they nest a `<button>` inside the `<td>`, perhaps for cross-browser operability.

```
      <td role="gridcell">
        <button
          type="button"
          tabindex="-1"
          aria-current="date"
          aria-pressed="false"
          aria-label="April 1, 2026">
          1
        </button>
      </td> 
```
